readthedocs/proxito/middleware.py

```diff
@@ -246,6 +246,19 @@ def add_hsts_headers(self, request, response):
             # See https://tools.ietf.org/html/rfc6797
             response['Strict-Transport-Security'] = '; '.join(hsts_header_values)
 
+    def add_cache_headers(self, request, response):
+        """
+        Add Cache-Control headers.
+
+        If private projects are allowed don't cache any request at the CDN level.
+        In the future we should set this header conditionally on private/public versions,
+        and on views with/without cookies (like the footer).
+
+        See https://developers.cloudflare.com/cache/about/cdn-cache-control.
+        """
+        if settings.ALLOW_PRIVATE_REPOS:
+            response['CDN-Cache-Control'] = 'private'
```

Comment:
Yea, this seems like a naive implementation, but fine for now 👍 

I do wonder about using `Cache-Control` headers instead of the CDN-specific ones. We should probably add the reason we're using those here -- is it because we're OK with users having a local cache of it, but just don't want the CDN to cache it?

Reply:
> I do wonder about using `Cache-Control` headers instead of the CDN-specific ones

If I understand correctly from the link that Santos put in the docstring, we can't use that one because it does not affect CF cache decisions. It says:

> Return the CDN-Cache-Control response header which Cloudflare evaluates to make caching decisions. Cache-Control, if also returned by the origin, is proxied as is and does not affect caching decisions made by Cloudflare. Additionally, CDN-Cache-Control is proxied downstream in case there are other CDNs between Cloudflare and the browser.

Reply:
yeah, `CDN-Cache-Control` applies to the shared cache only, `Cache-Control` is for the browser, which is fine if it caches things there.